cloud/datasync/example/index.ts

```diff
@@ -35,8 +35,8 @@ const router = middleware.createSyncExpressRouter();
 app.use('/', router);
 
 app.listen(3000, function() {
-  // tslint:disable-next-line:no-console
-  console.log('Example app listening on port 3000!');
+  logger.info('Example app listening on port 3000!',
+    {level: 'INFO', tag: 'cloud:datasync:example', src: 'index.ts'});
```

Comment:
I don't see the point of having the logger interface with `params: ...any[]` and just passing in parameters that are specific for the underlying library (bunyan it seems for this)

Now that we're utilizing the interface in the examples, I think we should refactor the common ground we see into the interface itself, something like

```typescript
interface LoggerData {
  tag?: string
  sourceFile?: string
}
class Logger { info(message:string, data: LoggerData) }
```

That way we can map that data to the way the underlying library accepts it:

```typescript
export class BunyanLogger implements Logger {
  info(msg:string, params:LoggerData) {
    this.bunyan.info(msg, { level: 'INFO', tag: params.tag, src: params.sourceFile })
  }
}
```